### DISCUSSION TITLE
How to display full category name in a bar chart or similar chart?

### DISCUSSION THREAD

**Discussion by lenard510** [vega/altair#3195] [General]
Need a way to show the full text per each category value of the category column. Right now as you can see in the screenshot, they are truncated to a limited size.  Also if possible, is there a way to manually set the chart area to the minimum, so that it is mostly the category bins in display?

![image](https://github.com/altair-viz/altair/assets/97851453/495fdc81-1aa8-4cd9-be4f-6eb67f1627f8)

Sample codes used for this screen. 

bar_chart1 = alt.Chart(df).transform_aggregate(
        count='count()',
        groupby=['Category']
    ).transform_joinaggregate(
        total='sum(count)'
    ).transform_calculate(
        frac=alt.datum.count / alt.datum.total

    ).mark_bar().encode(
        x=alt.X("count:Q",axis=None),
        y=alt.Y('Category').sort('-x')
    )

Thank you for your time.

**Comment by joelostblom** [comment]
You can use the `labelLimit` axis parameter to control the label length https://altair-viz.github.io/user_guide/generated/core/altair.Axis.html. The `width` and `height` parameters of the chart controls the overall chart area

**Reply by lenard510** [reply]
Hi @joelostblom Thank you for replying! This works! Is there a way to scroll through the labels if there are over 1000 different label values? The height is set to 400 fixed, so the more labels there are, the busies it looks. Just wondering if scrolling is supported for this use case.  Thank you.

**Reply by joelostblom** [reply]
I don't know of any way to scroll labels unfortunately.
